source/io/module_javaScriptInvoke.js

```diff
@@ -111,19 +105,32 @@ var assignJavaScriptInvoke;
          * Write JS value to a JS reference local
          * @param javaScriptValueRef VIA local for this JS reference
          * @param jsValue the JS value to associate with an existing or new cookie
+         * @param isStaticReference Whether the JS reference is a static reference.
+         * Static reference (Static control reference) shares cookie for the same jsValue.
+         * Dynamic reference (JS opaque reference from JSLI) always creates a new cookie even for the same jsValue.
          */
-        Module.javaScriptInvoke.writeJavaScriptRefNum = function (javaScriptValueRef, jsValue) {
-            var cachedCookie = getCachedRefNumCookie(jsValue);
-            if (cachedCookie !== undefined) {
-                Module.eggShell.writeDouble(javaScriptValueRef, cachedCookie); // set the VIA local to be this cookie value
-                return;
+        var writeJavaScriptStaticAndDynamicRefNum = function (javaScriptValueRef, jsValue, isStaticReference) {
+            if (isStaticReference) { // static reference (ie, control reference) shares cookie for the same jsValue
```

Comment:
We already discussed this. Please create a new Vireo API to check if the TypeRef in javaScriptValueRef is static or dynamic. That way you don't need to pass isStaticReference and you don't need two writeJavaScriptRefNum functions. I think clients don't need to know the difference.

Reply:
Done.